src/utilities/createRouteMethods.ts

```diff
@@ -30,16 +34,56 @@ export function createRouteMethods<T extends Routes>(routes: Resolved<Route>[]):
   return methods as any
 }
 
-function createCallableNode(route: Resolved<Route>): RouteMethod {
-  const node: RouteMethod = (values) => {
-    const url = assembleUrl(route, values)
-    const { then } = new Promise<Then<RouteMethodResponse>>(resolve => resolve({ url }))
+type CreateRouteMethodArgs = {
+  route: Resolved<Route>,
+  routerPush: RouterPush,
+}
+
+function createRouteMethod({ route, routerPush }: CreateRouteMethodArgs): RouteMethod {
```

Comment:
I thought the implementation here turned out nice and clean. Good sign that we're making a good choice.